pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -586,6 +587,24 @@ func (r *ReconcileBareMetalHost) actionProvisioning(prov provisioner.Provisioner
 		return actionContinueNoWrite{}
 	}
 
+	// Ensure the root device hints we're going to use are stored.
+	//
+	// If the user has provided explicit root device hints, they take
+	// precedence. Otherwise use the values from the hardware profile.
+	hintSource := info.host.Spec.RootDeviceHints
+	if hintSource == nil {
+		hwProf, err := hardware.GetProfile(info.host.HardwareProfile())
+		if err != nil {
+			return actionError{errors.Wrap(err, "Could not update root device hints")}
+		}
+		hintSource = &hwProf.RootDeviceHints
+	}
+	if (hintSource != nil && info.host.Status.RootDeviceHints == nil) || *hintSource != *(info.host.Status.RootDeviceHints) {
+		info.log.Info("updating root device hints")
+		info.host.Status.RootDeviceHints = hintSource
```

Comment:
Given that this only gets set when we reach provisioning status, I think we should also clear it when we deprovision so that it's only set when the host is provisioned (or provisioning).

I also wonder if we should in fact copy this, and any other provisioning-time configuration, to Status at the same time as we exit from the Ready state into the Provisioning state. It's really unclear what happens if you change it in the middle of provisioning, but a good bet is that the Status at the end of provisioning may or may not reflect what the host was actually provisioned with, depending on at what point during provisioning you changed it.

Reply:
I've updated the PR to clear the field on deprovision. I think the other question is probably best left to a separate PR.